vcd/resource_vcd_provider_vdc.go

```diff
@@ -0,0 +1,688 @@
+	}
+	networkPool, err := vcdClient.GetNetworkPoolById(networkPoolId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkPoolHref, err := networkPool.GetOpenApiUrl()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	providerVdcCreation := types.ProviderVdcCreation{
+		Name:                            providerVdcName,
+		Description:                     providerVdcDescription,
+		HighestSupportedHardwareVersion: hwVersion,
+		IsEnabled:                       d.Get("is_enabled").(bool),
+		VimServer: []*types.Reference{
+			{
+				HREF: vcenterUrl,
+				ID:   extractUuid(vcenter.VSphereVcenter.VcId),
+				Name: vcenter.VSphereVcenter.Name,
+			},
+		},
+		ResourcePoolRefs: &types.VimObjectRefs{
+			VimObjectRef: []*types.VimObjectRef{
+				{
+					VimServerRef: &types.Reference{
+						HREF: vcenterUrl,
+						ID:   extractUuid(vcenter.VSphereVcenter.VcId),
+						Name: vcenter.VSphereVcenter.Name,
+					},
+					MoRef:         resourcePool.ResourcePool.Moref,
+					VimObjectType: "RESOURCE_POOL",
+				},
+			},
+		},
+		StorageProfile: wantedStorageProfiles,
+		NsxTManagerReference: types.Reference{
+			HREF: nsxtManagers[0].HREF,
+			ID:   extractUuid(nsxtManagers[0].HREF),
+			Name: nsxtManagers[0].Name,
+		},
+		NetworkPool: types.Reference{
+			HREF: networkPoolHref,
+			Name: networkPool.NetworkPool.Name,
+			ID:   extractUuid(networkPool.NetworkPool.Id),
+			Type: networkPool.NetworkPool.PoolType,
+		},
+		AutoCreateNetworkPool: false,
+	}
+	providerVdc, err := vcdClient.CreateProviderVdc(&providerVdcCreation)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(providerVdc.VMWProviderVdc.ID)
+	return resourceVcdProviderVdcRead(ctx, d, meta)
+}
+
+func resourceVcdProviderVdcRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return genericResourceVcdProviderVdcRead(ctx, d, meta, "resource")
+}
+
+func genericResourceVcdProviderVdcRead(ctx context.Context, d *schema.ResourceData, meta interface{}, origin string) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	providerVdcName := d.Get("name").(string)
+	providerVdcId := d.Id()
+	var extendedProviderVdc *govcd.ProviderVdcExtended
+	var err error
+	if extractUuid(d.Id()) != "" {
+		extendedProviderVdc, err = vcdClient.GetProviderVdcExtendedById(providerVdcId)
+	} else {
+		extendedProviderVdc, err = vcdClient.GetProviderVdcExtendedByName(providerVdcName)
+	}
+	if err != nil {
+		log.Printf("[DEBUG] (%s) Could not find any extended Provider VDC with name %s: %s", origin, providerVdcName, err)
+		if origin == "datasource" {
+			return diag.Errorf("could not find any extended Provider VDC with name %s: %s", providerVdcName, err)
+		}
+		d.SetId("")
+		return nil
+	}
+	providerVdc, err := extendedProviderVdc.ToProviderVdc()
+	if err != nil {
+		log.Printf("[DEBUG] Could not find any Provider VDC with name %s: %s", providerVdcName, err)
+		return diag.Errorf("could not find any Provider VDC with name %s: %s", providerVdcName, err)
```

Comment:
Is the error message correct? Meaning - is it "finding" or "converting"? Can the `ProviderVdc` be not found if `ProviderVdcExtended` is found?

Reply:
Changed to a more precise message